.github/scripts/rollout_healthcheck.sh

```diff
@@ -0,0 +1,375 @@
+#!/bin/bash
+# Designed for use together with tools like Helm to provide a secondary health check on a release
+# in addition to default helm health checks. Ideally, you should set the timeout to be slightly less
+# than the default helm install/upgrade timeout to allow for this script to run and collect information from
+# ephemeral workloads before helm cleans them up in the event of an atomic failure.
+#
+# Note that this script does not surface log details, but does provide commands to fetch them if found
+#
+# Requirement: workloads to health check need the label 'app.kubernetes.io/name'
+#
+# Dependencies: oc, jq
+
+# ENV:
+# OC_NAMESPACE: namespace to scan
+# SKIP_AUTH: set to true to skip auth and use your existing local kubeconfig
+# OC_SERVER: OpenShift server URL
+# TIMEOUT_SECONDS: timeout in seconds for health check
+# POLL_INTERVAL_SECONDS: interval in seconds to poll health check
+# LABEL_SELECTOR: label selector to filter resources to health check on
+# ERROR_EXPR: error expression to search for in logs
+
+# TODO: break out funcs into plugins
+
+# configure defaults
+if [ -z "$ERROR_EXPR" ]; then
+    ERROR_EXPR="error|fatal|exception|stacktrace"
+fi
+if [ -z "$TIMEOUT_SECONDS" ]; then
+    TIMEOUT_SECONDS=420 # 7m
+fi
+if [ -z "$POLL_INTERVAL_SECONDS" ]; then
+    POLL_INTERVAL_SECONDS=15
+fi
+
+# will be set to 1 if a timeout occurs from an unfinished rollout
+# we use this to fetch triage info
+TIMED_OUT=0
+
+# string of commands to run after health check for additional info
+COMMANDS_TO_RUN=""
+
+# global flag to indicate if health check passed
+HEALTH_CHECK_PASSED=0
+
+
+set -e # failfast
+trap 'echo "Error occurred at line $LINENO while executing function $FUNCNAME"' ERR
+
+help_str() {
+    echo "Usage: SKIP_AUTH=true LABEL_SELECTOR=\"app.kubernetes.io/instance=nr-compliance-enforcement-PRNUM\" OC_NAMESPACE=c1c7ed-dev .github/scripts/rollout_healthcheck.sh"
```

Comment:
Should the script also contain checks for LABEL_SELECTOR and OC_NAMESPACE to validate they are passed in?